torch/csrc/jit/script/schema_type_parser.cpp

```diff
@@ -214,6 +215,29 @@ std::pair<TypePtr, c10::optional<AliasInfo>> SchemaTypeParser::parseType() {
       parseTensorDType(L.cur().text())) {
     value = parseRefinedTensor();
     alias_info = parseAliasAnnotation();
+  } else if (L.cur().kind == TK_IDENT && L.cur().text() == "__torch__") {
+    L.next();
+    L.expect('.');
+    auto torch_tok = L.expect(TK_IDENT);
+    if (torch_tok.text() != "torch") {
+      throw ErrorReport(torch_tok.range)
+          << "Expected classes namespace but got " << torch_tok.text();
+    }
+    L.expect('.');
+    auto classes_tok = L.expect(TK_IDENT);
+    if (classes_tok.text() != "classes") {
+      throw ErrorReport(classes_tok.range)
+          << "Expected classes namespace but got " << classes_tok.text();
+    }
+    L.expect('.');
+    auto class_tok = L.expect(TK_IDENT);
+    value = getCustomClass(
```

Comment:
We should write up a design for how namespaces work with custom classes. I think we want a unified namespace where user-defined classes live in the same namespace as everything else in the schema. Demoting them to `__torch__.torch.classes.` is weird because there is nothing valid in `__torch__` or `__torch__.torch`, and because it is going to be confusing to have a user defined class called Dict that is not the same class as Dict.